contrib/config/consul/config.go

```diff
@@ -0,0 +1,84 @@
+package consul
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/go-kratos/kratos/v2/config"
+	"github.com/hashicorp/consul/api"
+)
+
+// Option is etcd config option.
+type Option func(o *options)
+
+type options struct {
+	ctx  context.Context
+	path string
+}
+
+//  WithContext with registry context.
+func WithContext(ctx context.Context) Option {
+	return Option(func(o *options) {
+		o.ctx = ctx
+	})
+}
+
+// WithPath is config path
+func WithPath(p string) Option {
```

Comment:
WithNamespace?

Reply:
Etcd config use `withPath`。The consul config is like etcd，so all use `withPath` is reasonable。